### ISSUE TITLE
[v2.2.*] Giving your template files a name breaks the filter on templateselect 

### ISSUE THREAD

**Issue by PhillippOhlandt** [bolt/bolt#4577]
Hey,

I tested this on 2.2.13 (my current running version) and on 2.2.14 as well, both with the same bug/behavior.

What I did:

1: Installing a fresh copy of v2.2.13
2: Editing the templateselect filter on the `pages` contenttype
```
template:
    type: templateselect
    filter: 'page_*.twig'
```

3: Creating `page_1.twig` and `page_2.twig` in the `base-2014` theme.

Now the templateselect shows me only these two templates.

4: Adding the `templateselect` configuration to the themes `config.yml`
```
templateselect:
    templates:
        - name: 'Template for Records'
          filename: 'record.twig'
        - name: 'Page 1'
          filename: 'page_1.twig'
```

Now I can only select the `record.twig` and the `page_1.twig` (with their given names) on the templateselect. 
But I shouldnt be able to select the `record.twig` template and instead I should be able to select the `page_2.twig` template.

Seems like the filter will be ignored and all templates with a given name will be shown.

**Comment by SahAssar** [comment]
Yeah, ran into this a while ago but forgot to open an issue.

**Comment by rossriley** [comment]
I always assumed this was intended behaviour... If you take over the list in your theme then that replaces the default (which is to search for `*.twig`) with your approved list.

**Comment by SahAssar** [comment]
@rossriley That seems odd to me, since different templates can be meant for different contenttypes. Say you have the following four templates:
```
templateselect:
    templates:
        - filename: "page_gallery.twig"
          name: "Page with gallery"
        - filename: "page_contactform.twig"
          name: "Page with contactform"
        - filename: "entry_map.twig"
          name: "Blogpost with map"
        - filename: "entry_facebook.twig"
          name: "Blogpost with facebook comments"
```

Then I probably want `filter: 'page_*.twig'` in the pages contenttype and `filter: 'entry_*.twig'` in the entries contenttype.

Or is there some other way to restrict named templates to a single contenttype?

**Comment by PhillippOhlandt** [comment]
What @SahAssar said. I use it in that way.

**Comment by bobdenotter** [comment]
I think this is one of those typical "It's not wrong, but probably not what you would expect" type of things. 

In my experience this usually means a specific part of functionality is trying to do too much, and it's time too kill some darlings to make it less confusing. 

I think it would make sense that something defined in the theme's `theme.yml` takes precedence over the global `config.yml`, as it does with most other things. The simplest solution would be to just make it do that, document it, and close it off. 

However, the downside would be that you can't specify different files for different contenttypes anymore. 

Ideally we'd expand the syntax to something like this: 

```
templateselect:
    pages:
        filter: 'page_*.twig'
        files: 
            - filename: "page_gallery.twig"
              name: "Page with gallery"
            - filename: "page_contactform.twig"
              name: "Page with contactform"
    entries:
        filter: 'entry_*.twig'
        files: 
            - filename: "entry_map.twig"
              name: "Blogpost with map"
            - filename: "entry_facebook.twig"
              name: "Blogpost with facebook comments"
    default: 
        filter: 'record_*.twig'
```

But, that will require some work. Any volunteers? ;-)

**Comment by rarila** [comment]
All except @bobdenotter move one step back ;-)

**Comment by SahAssar** [comment]
Alright, so not really a bug... Moving two labels back ;)

**Comment by GawainLynch** [comment]
> But, that will require some work. Any volunteers? ;-)

Before we do… can we step back, take a close look at all these "bit" and maybe first flesh out a design document. There just seems to be too much illogical overlap in things and we have a major version bump coming up that could be a good time to "break stuff" :grin:

**Comment by bobdenotter** [comment]
Oh, certainly. absolutely not for 3.0.

**Comment by ToBe998** [comment]
I realy can't think of this as something other than a bug.

Even small websites will very likely have more than one contenttype that has variants. At the moment, you can only have variants with the "templatefields" mechanism. There already *is* the filter property that has been implemented specifically to make variantes of a contenttype work. And this feature is not working with names.
You just can't give a typical editor Templates without names as he will be scared for life and stop using Bolt. And if you give them names, they WILL select the wrong variant which will break your website.

So, making a nice new concept might be a nice thing for a future version. But this issue clearly is a bug and not an *interesting sideeffect*.

**Comment by GawainLynch** [comment]
Fixed in #5127
